project/JavalibIRCleaner.scala

```diff
@@ -598,8 +598,15 @@ final class JavalibIRCleaner(baseDirectoryURI: URI) {
         (enclosingClassName == TypedArrayBufferBridge || enclosingClassName == TypedArrayBufferBridgeMod)
       }
 
+      // TODO: it's not good to allow accessing the whole
+      // runtime package.
+      def isRunTimeLinkingInfoClass =
+        cls == RuntimePackage || cls == RuntimeLinkingInfoClass
```

Comment:
Unfortunately this is not something we can do. We really *cannot* leave traces of `scala.*` in the `javalib`, because the `javalib` artifact must work without the `scala-library` nor `scalajs-library`. This is not just a nice-to-have at this point; it's a hard technical requirement.

Either the `compiler` must collapse uses of these methods, or the `JavalibIRCleaner` must collapse them.

Reply:
Coming back to this, I think the correct approach is for the `JavalibIRCleaner` to hard-code replacements for the relevant methods of `scala.scalajs.LinkingInfo`.

Reply:
Yeah, thanks, hard-coded the replacements for the required methods of `scala.scalajs.LInkingInfo`

```scala
        // LinkingInfo
        // Must be sync the transformations with those in `scala.scalajs.LinkingInfo`
        case IntrinsicCall(LinkingInfoClass, `esVersionMethodName`, Nil) =>
          LinkTimeProperty("core/esVersion")(IntType)

        case IntrinsicCall(LinkingInfoClass, `isWebAssemblyMethodName`, Nil) =>
          LinkTimeProperty("core/isWebAssembly")(BooleanType)

        case IntrinsicCall(LinkingInfoClass, `linkerVersionMethodName`, Nil) =>
          LinkTimeProperty("core/linkerVersion")(StringType)
```